R/google_analytics.R

```diff
@@ -19,10 +19,42 @@ getGoogleProfile <- function(tokenFileId = ""){
       data.frame(googleAnalyticsR::ga_view(accountId, webPropertyId, viewId)) %>% dplyr::filter(effective == "READ_AND_ANALYZE") %>% dplyr::select(id, accountId, webPropertyId, timezone)
     })
     # Join the timezone column to the original data frame.
-    df <- df %>% dplyr::left_join(newdf, by = c("accountId" = "accountId", "webPropertyId" = "webPropertyId", "viewId" = "id"))
+    df <- df %>% dplyr::left_join(newdf, by = c("accountId" = "accountId", "webPropertyId" = "webPropertyId", "viewId" = "id")) %>% dplyr::select(accountId, accountName, webPropertyId, webPropertyName, viewId, viewName, timezone)
+  }
+  # get V4 Account
+  v4df <- googleAnalyticsR::ga_account_list("ga4")
+  if (nrow(v4df) > 0) {
+    v4newdf <- purrr::map_dfr(v4df$accountId, function(id){
+      # account looks like accounts/123456 so get rid of accounts/ to get account ID part. As for property, it looks like properties/123345 so remove properties/ to get property id.
+      getGoogleAnalyticsV4Property(id) %>% mutate(accountId = stringr::str_replace(parent, "accounts/", ""), webPropertyId = stringr::str_replace(name, "properties/", ""))
+    })
+    v4newdf <- v4newdf %>% dplyr::distinct(accountId, name, .keep_all = T)
+    v4df <- v4df %>% dplyr::left_join(v4newdf, by = c("accountId" = "accountId", "propertyId" = "webPropertyId")) %>% mutate(viewId = "", viewName = "") %>% dplyr::select(accountId, account_name, propertyId, property_name, viewId, viewName, timeZone) %>% dplyr::rename(webPropertyId = propertyId, webPropertyName = property_name, accountName = account_name, timezone = timeZone)
```

Comment:
mutate to align columns before bind_rows is probably unnecessary.

Reply:
removed unnecessary murate.